src/server/v2.0/handler/member.go

```diff
@@ -0,0 +1,168 @@
+	if err := m.RequireProjectAccess(ctx, projectNameOrID, rbac.ActionCreate, rbac.ResourceMember); err != nil {
+		return m.SendError(ctx, err)
+	}
+	if params.ProjectMember == nil {
+		return m.SendError(ctx, errors.BadRequestError(nil).WithMessage("the project member should provide"))
+	}
+	req, err := toMemberReq(params.ProjectMember)
+	if err != nil {
+		return m.SendError(ctx, err)
+	}
+	id, err := m.ctl.Create(ctx, projectNameOrID, *req)
+	if err != nil {
+		return m.SendError(ctx, err)
+	}
+	return operation.NewCreateProjectMemberCreated().
+		WithLocation(fmt.Sprintf("/api/v2.0/projects/%v/members/%d", projectNameOrID, id))
+}
+
+func toMemberReq(memberReq *models.ProjectMember) (*member.Request, error) {
+	data, err := json.Marshal(memberReq)
+	if err != nil {
+		return nil, err
+	}
+	var result member.Request
+	err = json.Unmarshal(data, &result)
+	if err != nil {
+		return nil, err
+	}
+	return &result, nil
+}
+
+func (m *memberAPI) DeleteProjectMember(ctx context.Context, params operation.DeleteProjectMemberParams) middleware.Responder {
+	projectNameOrID := parseProjectNameOrID(params.ProjectNameOrID, params.XIsResourceName)
+	if err := m.RequireProjectAccess(ctx, projectNameOrID, rbac.ActionDelete, rbac.ResourceMember); err != nil {
+		return m.SendError(ctx, err)
+	}
+	if params.Mid == 0 {
+		return m.SendError(ctx, errors.BadRequestError(nil).WithMessage("the project member id is required."))
+	}
+	err := m.ctl.Delete(ctx, projectNameOrID, int(params.Mid))
+	if err != nil {
+		return m.SendError(ctx, err)
+	}
+	return operation.NewDeleteProjectMemberOK()
+}
+
+func (m *memberAPI) GetProjectMember(ctx context.Context, params operation.GetProjectMemberParams) middleware.Responder {
+	projectNameOrID := parseProjectNameOrID(params.ProjectNameOrID, params.XIsResourceName)
+	if err := m.RequireProjectAccess(ctx, projectNameOrID, rbac.ActionRead, rbac.ResourceMember); err != nil {
+		return m.SendError(ctx, err)
+	}
+
+	if params.Mid == 0 {
+		return m.SendError(ctx, errors.BadRequestError(nil).WithMessage("the member id can not be empty!"))
+	}
+
+	member, err := m.ctl.Get(ctx, projectNameOrID, int(params.Mid))
+	if err != nil {
+		return m.SendError(ctx, err)
+	}
+	return operation.NewGetProjectMemberOK().WithPayload(toProjectMemberResp(member))
+}
+
+func (m *memberAPI) ListProjectMembers(ctx context.Context, params operation.ListProjectMembersParams) middleware.Responder {
+	projectNameOrID := parseProjectNameOrID(params.ProjectNameOrID, params.XIsResourceName)
+	if err := m.RequireProjectAccess(ctx, projectNameOrID, rbac.ActionList, rbac.ResourceMember); err != nil {
+		return m.SendError(ctx, err)
+	}
+	entityName := ""
+	if params.Entityname != nil {
+		entityName = *params.Entityname
+	}
+	query, err := m.BuildQuery(ctx, nil, nil, params.Page, params.PageSize)
+	if err != nil {
+		return m.SendError(ctx, err)
+	}
+	total, err := m.ctl.Count(ctx, projectNameOrID, query)
+	if err != nil {
+		return m.SendError(ctx, err)
+	}
```

Comment:
Return directly when the total is zero so that we can reduce a DB query.

Reply:
Updated